CMakeLists.txt

```diff
@@ -0,0 +1,539 @@
+    add_library(ssl STATIC ${COMMON_ssl_SRCS})
+
+    target_include_directories(ssl PRIVATE ${CMAKE_CURRENT_SOURCE_DIR} include)
+    target_compile_definitions(ssl PRIVATE  _REENTRANT OPENSSLDIR="/usr/local/ssl" ENGINESDIR="/usr/local/lib/engines-1.1" OPENSSL_PIC)
+    
+    set(crypto_SRCS
+        ${COMMON_crypto_SRCS}
+        crypto/bf/bf_enc.c crypto/camellia/cmll_cbc.c crypto/sha/keccak1600.c
+        crypto/mem_clr.c crypto/camellia/cmll_misc.c crypto/whrlpool/wp_block.c
+        crypto/rc4/rc4_skey.c crypto/rc4/rc4_enc.c crypto/chacha/chacha_enc.c
+        crypto/aes/aes_core.c crypto/bn/bn_asm.c crypto/des/fcrypt_b.c
+        crypto/camellia/camellia.c crypto/des/des_enc.c
+    )
+
+    add_library(crypto STATIC ${crypto_SRCS})
+
+    target_include_directories(crypto PRIVATE crypto/ec/curve448/arch_32 crypto/include include ${CMAKE_CURRENT_SOURCE_DIR} crypto/modes crypto/ec/curve448 crypto)
+    target_compile_definitions(crypto PRIVATE  _REENTRANT OPENSSLDIR="/usr/local/ssl" ENGINESDIR="/usr/local/lib/engines-1.1" OPENSSL_PIC)
+    set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -mios-version-min=7.0.0 -fno-common -fPIC ")
+    set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -mios-version-min=7.0.0 -fno-common -fPIC ")
+        
+elseif(ANDROID)
+    IF(${ANDROID_ABI} STREQUAL "x86_64")
+
+        set(arch "android-x86_64")
+        message("Building ${arch}")
+        
+        file(COPY android/${arch}/opensslconf.h DESTINATION ${CMAKE_CURRENT_SOURCE_DIR}/include/openssl/)
+        file(COPY android/${arch}/bn_conf.h android/${arch}/dso_conf.h DESTINATION ${CMAKE_CURRENT_SOURCE_DIR}/crypto/include/internal/)
+        file(COPY android/${arch}/buildinf.h DESTINATION ${CMAKE_CURRENT_SOURCE_DIR}/crypto/)
+            
+        add_library(ssl STATIC ${COMMON_ssl_SRCS})
+
+        target_include_directories(ssl PRIVATE ${CMAKE_CURRENT_SOURCE_DIR} include)
+        target_compile_definitions(ssl PRIVATE OPENSSL_USE_NODELETE OPENSSL_PIC ENGINESDIR="/usr/local/lib/engines-1.1" OPENSSLDIR="/usr/local/ssl")
+        
+        set(crypto_SRCS
+            ${COMMON_crypto_SRCS}
+            crypto/engine/eng_all.c crypto/rc4/rc4_enc.c crypto/camellia/camellia.c
+            crypto/engine/eng_rdrand.c crypto/engine/tb_cipher.c crypto/camellia/cmll_cbc.c
+            crypto/engine/eng_cnf.c crypto/engine/eng_openssl.c crypto/engine/tb_rand.c
+            crypto/engine/eng_err.c crypto/engine/tb_eckey.c crypto/engine/eng_init.c
+            crypto/chacha/chacha_enc.c crypto/engine/eng_lib.c crypto/engine/eng_dyn.c
+            crypto/engine/eng_ctrl.c crypto/engine/eng_list.c crypto/engine/tb_pkmeth.c
+            crypto/engine/eng_pkey.c crypto/sha/keccak1600.c crypto/engine/tb_rsa.c
+            crypto/engine/tb_dsa.c crypto/mem_clr.c crypto/bn/bn_asm.c
+            crypto/engine/tb_dh.c crypto/engine/tb_asnmth.c crypto/engine/eng_table.c
+            crypto/engine/eng_fat.c crypto/whrlpool/wp_block.c crypto/engine/tb_digest.c
+            crypto/rc4/rc4_skey.c
```

Comment:
if you were to sort these sources for all android architectures, shouldn't they be the same?

Reply:
Probably, I only did set intersection between the common one and the rest. I'll check.

Reply:
Yeah, you're right. They're the same sans one source file for x86_64 I think. Moreover, iOS Device flavor has the same sources as Android. I'll carry them out to a separate variable then.

Reply:
👍